datalad/support/gitrepo.py

```diff
@@ -526,8 +526,15 @@ def git_get_remote_branches(self):
         # For some reason, this is three times faster than the version below:
         remote_branches = list()
         for remote in self.repo.remotes:
-            for ref in remote.refs:
-                remote_branches.append(ref.name)
+            try:
+                for ref in remote.refs:
+                    remote_branches.append(ref.name)
+            except AssertionError as e:
+                if e.message.endswith("did not have any references"):
+                    # this will happen with git annex special remotes
+                    pass
+                else:
+                    raise e
```

Comment:
This try block is per remote -- either a remote is special or not. Manual test had both types of remotes. My test repo creation foo is not yet advanced enough to produce such a test. I'll keep praying for enlightenment... ;-)